code/modules/mob/living/life.dm

```diff
@@ -197,6 +198,25 @@
 		if(!remote_view && !client.adminobs)
 			reset_perspective(null)
 
+/mob/living/proc/handle_darksight(var/no_adjustment_scaling = FALSE)
+	var/obj/screen/fullscreen/see_through_darkness/S = screens["see_through_darkness"]
+	if(!S)
+		return
+	var/current = S.alpha / 255		//Our current adjustedness.
+	var/brightness = 0.0 			//We'll assume it's superdark if we can't find something else..
+	if(isturf(loc))
+		var/turf/T = loc			//Will be true 99% of the time, thus avoiding the whole elif chain.
+		brightness = T.get_lumcount()
+	var/darkness = 1 - brightness	//Silly, I know, but 'alpha' and 'darkness' go the same direction on a number line.
+
+	var/adjust_to = no_adjustment_scaling ? darkness : min(darkness, min(see_in_dark / world.view, 1.0)) /*Capped by how darksighted they are if this proc is allowed to go all out.
+															The right-side argument of min() is a ratio of how good your darksight is, from 'nada' to 'really darn good'.*/
+	var/distance = abs(current - adjust_to) //Used for how long to animate for.
+	if(distance < 0.01)				//We're already all set.
+		return
+
+	animate(S, alpha = (adjust_to * 255), time = (distance * 10 SECONDS)) //Vite in umbra.
```

Comment:
and `animate` is scary, especially when you start dealing with images that cover your entire screen.

`animate` is a horrible abomination whereby it chops up the image, server side, creates the animation, renders it all, then pushes out said image to every single client in the world (even if they can't see it). It's prone to causing crashing too; it's why our very first iteration of baylights caused horrifying crashes and terrible experiences for people.

Reply:
This is unfortunately the pivotal mechanic allowing eye adjustment to be dynamic and is made necessary by the varying alpha and time to adjust

Reply:
It isn't though.

Reply:
I'd love to hear your idea, it sounds like you have something very interesting in mind

Reply:
Yeah, its rate is adjusted based on just how different the brightness of the area is to how adjusted the eyes are already. I'm not sure more it'd synchronize to. Reading the DMM 'the actual interpolation between frames is all done on the client.' snippet for animation makes a bit of sense to me out of why it could cause crashing

This also means that your eyes will only get so adjusted to the dark given the brightness of the area, so the only time you will ever have full darksight is in full darkness